pkg/schema/v1/replica_set.go

```diff
@@ -119,6 +123,46 @@ func (r *ReplicaSet) Obtain(k8s kmetav1.Object) {
 	}
 }
 
+func (r *ReplicaSet) getIcingaState() (IcingaState, string) {
+	if r.DesiredReplicas < 1 {
+		reason := fmt.Sprintf("ReplicaSet %s/%s has an invalid desired replica count: %d", r.Namespace, r.Name, r.DesiredReplicas)
+
+		return Unknown, reason
+	}
+
+	gracePeriodReason := IsWithinGracePeriod(r)
+	if gracePeriodReason != nil {
+		return Ok, *gracePeriodReason
+	}
+
+	for _, condition := range r.Conditions {
+		if condition.Type == "ReplicaFailure" && condition.Status == "True" {
+			reason := fmt.Sprintf("ReplicaSet %s/%s has a failure condition: %s", r.Namespace, r.Name, condition.Message)
+
+			return Critical, reason
+		}
+	}
+
+	switch {
+	case r.AvailableReplicas < 1:
+		reason := fmt.Sprintf("ReplicaSet %s/%s has no available replicas", r.Namespace, r.Name)
+
+		return Critical, reason
+	case r.ReadyReplicas < r.DesiredReplicas:
+		reason := fmt.Sprintf("The number of ready replicas is less than the number of desired replicas: %d current/%d desired", r.ReadyReplicas, r.DesiredReplicas)
```

Comment:
Please rephrase:

`ReplicaSet %s/%s only has %d out of %d desired replicas available.`